content/en/docs/tasks/security/authorization/authz-ingress/index.md

```diff
@@ -46,128 +39,383 @@ original client source IP on the ingress gateway using the following command:
-the original client source IP address, which will be used in the authorization policy:
-
-    {{< text bash >}}
-    $ CLIENT_IP=$(curl "$INGRESS_HOST":"$INGRESS_PORT"/ip -s | grep "origin" | cut -d'"' -f 4) && echo "$CLIENT_IP"
-    105.133.10.12
-    {{< /text >}}
-
 {{< warning >}}
 If you don’t see the expected output, retry after a few seconds.
 Caching and propagation overhead can cause a delay.
 {{< /warning >}}
 
+## Getting traffic into Kubernetes and Istio
+
+All methods of getting traffic into Kubernetes involve opening a port on all worker nodes.  The main features that accomplish this are the `NodePort` service and the `LoadBalancer` service.  Even the Kubernetes `Ingress` resource must be backed by an Ingress controller that will create either a `NodePort` or a `LoadBalancer` service.
+
+* A `NodePort` just opens up a port in the range 30000-32767 on each worker node and uses a label selector to identify which Pods to send the traffic to.  You have to manually create some kind of load balancer in front of your worker nodes or use Round-Robin DNS.
+
+* A `LoadBalancer` is just like a `NodePort`, except it also creates an environment specific external load balancer to handle distributing traffic to the worker nodes.  For example, in AWS EKS, the `LoadBalancer` service will create a Classic ELB with your worker nodes as targets. If your Kubernetes environment does not have a `LoadBalancer` implementation, then it will just behave like a `NodePort`.  An Istio ingress gateway creates a `LoadBalancer` service.
+
+What if the Pod that is handling traffic from the `NodePort` or `LoadBalancer` isn't running on the worker node that received the traffic?  Kubernetes has its own internal proxy called kube-proxy that receives the packets and forwards them to the correct node.
+
+## Source IP address of the original client
+
+If a packet goes through an external proxy load balancer and/or kube-proxy, then the original source IP address of the client is lost.  Below are some strategies for preserving the original client IP for logging or security purposes.
+
+{{< tabset category-name="lb" >}}
+
+{{< tab name="TCP/UDP Proxy Load Balancer" category-value="proxy" >}}
+
+If you are using a TCP/UDP Proxy external load balancer (AWS Classic ELB), it can use the [Proxy Protocol](https://www.haproxy.com/blog/haproxy/proxy-protocol/) to embed the original client IP address in the packet data.  Both the external load balancer and the Istio ingress gateway must support the proxy protocol for it to work.  In Istio, you can enable it with an `EnvoyFilter` like below:
+
+{{< text yaml >}}
+apiVersion: networking.istio.io/v1alpha3
+kind: EnvoyFilter
+metadata:
+  name: proxy-protocol
+  namespace: istio-system
+spec:
+  workloadSelector:
+    labels:
+      istio: ingressgateway
+  configPatches:
+  - applyTo: LISTENER
+    patch:
+      operation: MERGE
+      value:
+        listener_filters:
+        - name: envoy.listener.proxy_protocol
```

Comment:
```
        - name: envoy.listener.tls_inspector
```

Is not needed?

Reply:
It's not needed just to enable the proxy protocol.  If you already have it in the config, the EnvoyFilter above won't remove it.  I'm not sure whether they are compatible or not though.

Reply:
Without `envoy.listener.tls_inspector` `https://` does not work; maybe add it as note? Also its mentioned in here https://github.com/istio/istio.io/blob/master/content/en/docs/ops/configuration/traffic-management/network-topologies/index.md#proxy-protocol and in here https://github.com/istio/istio/pull/12748#issuecomment-640274792.

Reply:
I didn't notice they had also added Proxy protocol documentation to the Network Topologies doc, but you're right the two EnvoyFilters there and here should probably match.

Reply:
FYI, PR here: https://github.com/istio/istio.io/pull/8472

Reply:
Perfect; thank you. Much appreciated.